--- a/Makefile.in
+++ b/Makefile.in
@@ -2846,6 +2846,8 @@ do-check:
 	@: $(MAKE); $(unstage)
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
+	echo "Unset ELF_PACKAGE_METADATA for tests"; \
+	export -n ELF_PACKAGE_METADATA; \
 	$(MAKE) $(RECURSE_FLAGS_TO_PASS) check-host check-target
 
 # Automated reporting of test results.
@@ -4353,6 +4355,7 @@ check-bfd:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/bfd && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -5493,6 +5496,7 @@ check-opcodes:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/opcodes && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -6633,6 +6637,7 @@ check-binutils:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/binutils && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -7558,6 +7563,7 @@ check-cgen:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/cgen && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -8026,6 +8032,7 @@ check-dejagnu:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/dejagnu && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -8494,6 +8501,7 @@ check-etc:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/etc && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -10119,6 +10127,7 @@ check-fixincludes:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/fixincludes && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -11699,6 +11708,7 @@ check-gas:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gas && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -12848,6 +12858,7 @@ check-gcc:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gcc && \
 	  $(MAKE) $(FLAGS_TO_PASS) $(EXTRA_GCC_FLAGS) $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -13997,6 +14008,7 @@ check-gmp:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gmp && \
 	  $(MAKE) $(FLAGS_TO_PASS) AM_CFLAGS="-DNO_ASM" $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -15134,6 +15146,7 @@ check-mpfr:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/mpfr && \
 	  $(MAKE) $(FLAGS_TO_PASS) AM_CFLAGS="-DNO_ASM" $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -16271,6 +16284,7 @@ check-mpc:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/mpc && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -17408,6 +17422,7 @@ check-isl:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/isl && \
 	  $(MAKE) $(FLAGS_TO_PASS) V=1 $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -18536,6 +18551,7 @@ check-gold:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gold && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -18990,6 +19006,7 @@ check-gprof:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gprof && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -19458,6 +19475,7 @@ check-gprofng:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gprofng && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -20621,6 +20639,7 @@ check-gettext:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gettext && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -20924,6 +20943,7 @@ check-tcl:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/tcl && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -21377,6 +21397,7 @@ check-itcl:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/itcl && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -22531,6 +22552,7 @@ check-ld:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/ld && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -23671,6 +23693,7 @@ check-libbacktrace:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libbacktrace && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -24811,6 +24834,7 @@ check-libcpp:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libcpp && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -25951,6 +25975,7 @@ check-libcody:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libcody && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -26940,6 +26965,7 @@ check-libdecnumber:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libdecnumber && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -27379,6 +27405,7 @@ check-libgui:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libgui && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -28542,6 +28569,7 @@ check-libiberty:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libiberty && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -29691,6 +29719,7 @@ check-libiberty-linker-plugin:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libiberty-linker-plugin && \
 	  $(MAKE) $(FLAGS_TO_PASS) @extra_linker_plugin_flags@ $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -30840,6 +30869,7 @@ check-libiconv:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libiconv && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -31189,6 +31219,7 @@ check-m4:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/m4 && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -31657,6 +31688,7 @@ check-readline:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/readline && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -32125,6 +32157,7 @@ check-sid:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/sid && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -32593,6 +32626,7 @@ check-sim:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/sim && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -33061,6 +33095,7 @@ check-texinfo:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/texinfo && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -34648,6 +34683,7 @@ check-gnulib:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gnulib && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -35116,6 +35152,7 @@ check-gdbsupport:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gdbsupport && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -35584,6 +35621,7 @@ check-gdbserver:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gdbserver && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -36052,6 +36090,7 @@ check-gdb:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gdb && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -36520,6 +36559,7 @@ check-expect:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/expect && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -36988,6 +37028,7 @@ check-guile:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/guile && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -37456,6 +37497,7 @@ check-tk:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/tk && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -38788,6 +38830,7 @@ check-c++tools:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/c++tools && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -39240,6 +39283,7 @@ check-gnattools:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gnattools && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -40403,6 +40447,7 @@ check-lto-plugin:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/lto-plugin && \
 	  $(MAKE) $(FLAGS_TO_PASS) @extra_linker_plugin_flags@ $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -40857,6 +40902,7 @@ check-libcc1:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libcc1 && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -41325,6 +41371,7 @@ check-gotools:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/gotools && \
 	  $(MAKE) $(FLAGS_TO_PASS)  check)
 
@@ -42479,6 +42526,7 @@ check-libctf:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libctf && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -43619,6 +43667,7 @@ check-libsframe:
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) $(EXTRA_HOST_EXPORTS) \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libsframe && \
 	  $(MAKE) $(FLAGS_TO_PASS)  $(EXTRA_BOOTSTRAP_FLAGS) check)
 
@@ -44351,6 +44400,7 @@ all-libgrust: configure-libgrust
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS)  \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/libgrust && \
 	  $(MAKE) $(BASE_FLAGS_TO_PASS) $(EXTRA_HOST_FLAGS) $(STAGE1_FLAGS_TO_PASS)  \
 		$(TARGET-libgrust))
--- a/Makefile.tpl
+++ b/Makefile.tpl
@@ -1012,6 +1012,8 @@ do-check:
 	@: $(MAKE); $(unstage)
 	@r=`${PWD_COMMAND}`; export r; \
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
+	echo "Unset ELF_PACKAGE_METADATA for tests"; \
+	export -n ELF_PACKAGE_METADATA; \
 	$(MAKE) $(RECURSE_FLAGS_TO_PASS) check-host check-target
 
 # Automated reporting of test results.
@@ -1393,6 +1395,7 @@ check-[+module+]:
 	s=`cd $(srcdir); ${PWD_COMMAND}`; export s; \
 	$(HOST_EXPORTS) [+ IF bootstrap +]$(EXTRA_HOST_EXPORTS)[+
 	ENDIF bootstrap +] \
+	export -n ELF_PACKAGE_METADATA; \
 	(cd $(HOST_SUBDIR)/[+module+] && \
 	  $(MAKE) $(FLAGS_TO_PASS) [+extra_make_flags+][+
 	  IF bootstrap +] $(EXTRA_BOOTSTRAP_FLAGS)[+ ENDIF bootstrap +] check)
--- a/ld/emultempl/elf.em
+++ b/ld/emultempl/elf.em
@@ -100,8 +100,15 @@ fragment <<EOF
   link_info.error_execstack = DEFAULT_LD_ERROR_EXECSTACK;
   link_info.warn_is_error_for_rwx_segments = DEFAULT_LD_ERROR_RWX_SEGMENTS;
   link_info.memory_seal = DEFAULT_LD_Z_MEMORY_SEAL;
-}
 
+  char *package_metadata = getenv("ELF_PACKAGE_METADATA");
+  if (package_metadata) {
+    char * const p = xmalloc (strlen (package_metadata) + 1);
+    strcpy (p, package_metadata);
+    ldelf_emit_note_fdo_package_metadata = (const char *)p;
+  }
+}
+ 
 EOF
 fi
 
