Author: James Addison <jay@jp-hosting.net>
Description: Generate a deterministic gprofng example documentation tarball
Bug-Debian: https://bugs.debian.org/1090395
Bug-Debian: https://bugs.debian.org/1092870
Last-Update: 2025-01-13

--- a/gprofng/doc/Makefile.am
+++ b/gprofng/doc/Makefile.am
@@ -63,7 +63,7 @@ EXTRA_DIST = $(man_MANS) version.texi
 info: $(man_MANS)
 
 examples.tar.gz:
-	$(AM_V_at)( tar czf $@ $(srcdir)/../examples )
+	$(AM_V_at)( tar czf $@ --sort=name --mtime="@${SOURCE_DATE_EPOCH}" --owner=0 --group=0 --numeric-owner --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime --mode=a=rX,u+w $(srcdir)/../examples )
 
 .PHONY: install-data-local
 
--- a/gprofng/doc/Makefile.in
+++ b/gprofng/doc/Makefile.in
@@ -905,7 +905,7 @@ gprofng-display-gmon.1: $(srcdir)/gprofn
 info: $(man_MANS)
 
 examples.tar.gz:
-	$(AM_V_at)( tar czf $@ $(srcdir)/../examples )
+	$(AM_V_at)( tar czf $@ --sort=name --mtime="@${SOURCE_DATE_EPOCH}" --owner=0 --group=0 --numeric-owner --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime --mode=a=rX,u+w $(srcdir)/../examples )
 
 .PHONY: install-data-local
 
