--- a/gprofng/configure.ac
+++ b/gprofng/configure.ac
@@ -214,24 +214,22 @@ AM_ZLIB
 # Generate manpages, if possible.
 build_man=false
 build_doc=false
-if test $cross_compiling = no; then
-  AC_CHECK_PROGS([MAKEINFO], makeinfo, [$MISSING makeinfo])
-  case "x$MAKEINFO" in
-    x | */missing\ makeinfo*)
-      AC_MSG_WARN([gprofng: makeinfo is missing. Info documentation will not be built.])
-      ;;
-    *)
-      case x"`$MAKEINFO --version | grep 'GNU texinfo'`" in
-	x*\ [[1-5]].*|x*\ 6.[[0-4]].* )
-	  AC_MSG_WARN([gprofng: $MAKEINFO is too old. Info documentation will not be built.])
-	  MAKEINFO="@echo $MAKEINFO is too old, 6.5 or newer required; true"
-	  ;;
-	x* ) build_doc=true ;;
-	esac
-      ;;
-    esac
-  AC_SUBST(MAKEINFO)
-fi
+AC_CHECK_PROGS([MAKEINFO], makeinfo, [$MISSING makeinfo])
+case "x$MAKEINFO" in
+  x | */missing\ makeinfo*)
+    AC_MSG_WARN([gprofng: makeinfo is missing. Info documentation will not be built.])
+    ;;
+  *)
+    case x"`$MAKEINFO --version | grep 'GNU texinfo'`" in
+      x*\ [[1-5]].*|x*\ 6.[[0-4]].* )
+        AC_MSG_WARN([gprofng: $MAKEINFO is too old. Info documentation will not be built.])
+        MAKEINFO="@echo $MAKEINFO is too old, 6.5 or newer required; true"
+        ;;
+      x* ) build_doc=true ;;
+      esac
+    ;;
+  esac
+AC_SUBST(MAKEINFO)
 AM_CONDITIONAL([BUILD_MAN], [test x$build_man = xtrue])
 AM_CONDITIONAL([BUILD_DOC], [test x$build_doc = xtrue])
 
--- a/gprofng/configure
+++ b/gprofng/configure
@@ -17448,7 +17448,7 @@ fi
 # Generate manpages, if possible.
 build_man=false
 build_doc=false
-if test $cross_compiling = no; then
+if :; then
   for ac_prog in makeinfo
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
